modules/cas_disk/exp_obj.c

```diff
@@ -691,36 +722,29 @@ int casdsk_exp_obj_lock(struct casdsk_disk *dsk)
 
 	exp_obj = dsk->exp_obj;
 
-	exp_obj->locked_bd = cas_bdget_disk(exp_obj->gd);
-	if (!exp_obj->locked_bd)
-		return -ENAVAIL;
-
-	mutex_lock(&exp_obj->locked_bd->EXP_OBJ_MUTEX);
-
-	if (exp_obj->locked_bd->bd_openers) {
-		printk(CASDSK_KERN_DEBUG "Device %s in use (openers=%d). Refuse to stop\n",
-		       exp_obj->locked_bd->bd_disk->disk_name,
-		       exp_obj->locked_bd->bd_openers);
+	mutex_lock(&dsk->openers_lock);
 
+	if (dsk->openers > 0) {
+		mutex_unlock(&dsk->openers_lock);
 		casdsk_exp_obj_unlock(dsk);
 		return -EBUSY;
 	}
 
-	return 0;
+	dsk->claimed = true;
+	mutex_unlock(&dsk->openers_lock);
```

Comment:
How about adding wrappers for claiming and disclaiming the exported object?

Reply:
Basically, cas_exp_obj_lock/unlock are wrappers for claiming and disclaiming the exported object.